kartothek/io_components/utils.py

```diff
@@ -111,7 +112,7 @@ def _instantiate_store(store):
 
 
 def _ensure_compatible_indices(dataset, secondary_indices):
-    if dataset:
+    if dataset.exists:
```

Comment:
Is there a test of this?

Reply:
This is used in all writing functions where we migrated to the factory model by using `_ensure_factory`. 
Explicit unit test doesn't exist. Hell breaks loose if I change this, though. 

I would consider it _tested_ but I can add more tests if you think it is necessary. If so, do you think a unit test is sufficient or something larger?

Reply:
So why did this work before your PR then?